network/network_linux.go

```diff
@@ -289,12 +289,21 @@ func (nm *networkManager) addDNSServers(ifName string, dnsServers []string) (str
 		return osVersion, err
 	}
 
+	if len(dnsServers) == 0 {
```

Comment:
Adding this check again because of the caller behavior changes then this could be a potential point of failure.